### PR TITLE
Fix overwrite

### PR DESCRIPTION
**Pull request by gabaldon**

### CODE REVIEW

File: src/background/utils/overwriteWitnetNodeConfiguration.ts [hunk 1]
```diff
@@ -1,30 +1,60 @@
...
   sheikahPath,
   witnetConfigFileName,
   publicNodeUrls,
+  oldWitnetNodes,
 }: {
   sheikahPath: string
   witnetConfigFileName: string
   publicNodeUrls: Array<string>
+  oldWitnetNodes: Array<string>
 }) {
```
Comment (Tommytrg) [inline src/background/utils/overwriteWitnetNodeConfiguration.ts:30]: We are passing always the same arguments to this function. I like being able to pass them to be able to test the function easily, but could we add a default argument value?

```suggestion
const defaultNodeConfiguration = {
    sheikahPath: SHEIKAH_PATH,
    witnetConfigFileName: WITNET_CONFIG_FILE_NAME,
    publicNodeUrls: URLS_PUBLIC_WITNET_NODES,
    oldWitnetNodes: OLD_WITNET_NODE_IPS,
}

export function overwriteWitnetNodeConfiguration({
  sheikahPath,
  witnetConfigFileName,
  publicNodeUrls,
  oldWitnetNodes,
}: {
  sheikahPath: string
  witnetConfigFileName: string
  publicNodeUrls: Array<string>
  oldWitnetNodes: Array<string>
} = defaultNodeConfiguration) {
```

File: src/background/utils/overwriteWitnetNodeConfiguration.ts [hunk 2]
```diff
@@ -1,30 +1,60 @@
...
+
+export function hasOldWitnetIp({
+  oldWitnetNodes,
+  currentNodeConfiguration,
+}: {
+  oldWitnetNodes: Array<string>
+  currentNodeConfiguration: Array<string>
+}) {
+  return currentNodeConfiguration.some(ip => oldWitnetNodes.includes(ip))
+}
```
Comment (Tommytrg) [inline src/background/utils/overwriteWitnetNodeConfiguration.ts:17]: This function doesn't need to know about our business  logic
```suggestion
export function existIntersection(
  arr1: Array<string>,
  arr2: Array<string>,
)) {
  return arr1.some(item => arr2.includes(item))
}
```

File: src/background/utils/overwriteWitnetNodeConfiguration.ts [hunk 3]
```diff
@@ -1,30 +1,60 @@
...
+  oldWitnetNodes,
 }: {
   sheikahPath: string
   witnetConfigFileName: string
   publicNodeUrls: Array<string>
+  oldWitnetNodes: Array<string>
 }) {
   const replacement = `node_url = ${JSON.stringify(publicNodeUrls)}\n`
     .replace("'", '')
     .trim()
```
Comment (tmpolaczyk) [inline src/background/utils/overwriteWitnetNodeConfiguration.ts:33]: No need to calculate this string if hasOldWitnetIp returns false, so let's move it inside the if.

File: tests/unit/src/background/utils/overwriteWitnetNodeConfiguration.spec.ts [hunk 4]
```diff
@@ -28,20 +29,17 @@ describe('overwriteConfigFile', () => {
     )
   })
 
-  it('Should overwrite the configuration with several nodes according to the parameters given', () => {
+  it('Should overwrite config only if there is only one old witnet node ip', () => {
```
Comment (tmpolaczyk) [inline tests/unit/src/background/utils/overwriteWitnetNodeConfiguration.spec.ts:32]:
```suggestion
  it('Should overwrite config if there is only one old witnet node ip', () => {
```

**Review by tmpolaczyk — APPROVED** [review]

**Review by Tommytrg — APPROVED** [review]

**Pull request merged**